src/htmx.js

```diff
@@ -3008,26 +3008,35 @@ return (function () {
             var splitPath = path.split("#");
             var pathNoAnchor = splitPath[0];
             var anchor = splitPath[1];
-            var finalPathForGet = null;
-            if (verb === 'get' || verb === 'delete') {
-                finalPathForGet = pathNoAnchor;
+
+            // Encode body within the URL if:
+            // 1. The HTTP method for the request is GET
+            // 2. an extension on the element defines the `encodeParameters` function
+            var shouldEncodeBodyInUrl = verb === 'get'
+            withExtensions(elt, function(extension) {
+                if (typeof extension.encodeParameters === 'function') {
+                    shouldEncodeBodyInUrl = false
+                }
+            })
+
+            var finalPath = path
+            if (shouldEncodeBodyInUrl) {
+                finalPath = pathNoAnchor;
```

Comment:
what do you think about handling this via a new field on `requestConfig` above?  Something like `requestConfig.useBody`?  We could default that value to `verb == 'get'` and then let folks override it when necessary?

for the default we could also introduce a config variable, something like `htmx.config.paramRequestTypes` that defaults to `['get']` and test membership in that, which would allow folks who want correct delete behavior to write this:

```js
htmx.config.paramRequestTypes.push('delete');
```

Reply:
Agreed, except I did `requestConfig.useUrlParams` instead since I think it's a little more descriptive.

Reply:
Likewise I think `htmx.config.httpMethodsThatDefaultToUrlParams` instead of `htmx.config.paramRequestTypes`. I know it's a mouthful but it's not going to be a very common config (soon) and I'd rather the name be too long than too short.

Reply:
If you disagree on either of those names let me know and I'll change them.

Reply:
I have settled on `htmx.config.methodsThatUseUrlParams`